src/main/java/upeldev/com/github/upel3/repositories/GradeRepository.java

```diff
@@ -0,0 +1,29 @@
+package upeldev.com.github.upel3.repositories;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+import upeldev.com.github.upel3.model.Grade;
+import upeldev.com.github.upel3.model.User;
+
+import java.util.List;
+
+public interface GradeRepository extends CrudRepository<Grade, Integer> {
+
+    Grade findGradeById(@Param("activityId")  Long activityId);
+
+    List<Grade> findGradeByUser(User user);
+
+    @Query(value = "SELECT ig FROM Grade ig " +
+            "INNER JOIN Activity a on a.id = ig.activity.id " +
+            "INNER JOIN Course c on c.id = a.course.id " +
+            "WHERE c.id = :courseId")
+    List<Grade> findGradeByCourse(@Param("courseId") Long courseId);
+
+    @Query(value = "SELECT ig FROM Grade ig " +
+            "INNER JOIN Activity a on a.id = ig.activity.id " +
+            "INNER JOIN Course c on c.id = a.course.id " +
+            "WHERE c.id = :courseId and ig.user.id = :userId")
+    List<Grade> findGradeByCourseAndUser(@Param("courseId") Long courseId, @Param("userId") Long userId);
```

Comment:
I've tried it without sql, and this approach works very well with simple queries, but with those it wouldnt work, so i had to write SQL (or maybe there is some other method I didnt't know about) :/

Reply:
Well, it's very strange :| Maybe this can be caused by some problems with equals or hashcode methods, but it need to be checked. For now this can stay like it is, with SQL queries, and alternatively we'll work it out later

Reply:
It might be avoidable after adding the both-ways relation between Course and User. I'll look into that